File: engine/model/src/main/scala/net/server/channel/packet/special/MonsterMagnetPacket.scala
package net.server.channel.packet.special

import java.awt.Point

class MonsterMagnetPacket(private var _skillId: Int, private var _skillLevel: Int, private var _position: Point, private var _monsterData: Array[MonsterMagnetData], private var _direction: Byte) extends BaseSpecialMovePacket(_skillId = _skillId, _skillLevel = _skillLevel, _position = _position) {
  def monsterData: Array[MonsterMagnetData] = _monsterData

  def direction: Byte = _direction
}

File: engine/engine-model/src/main/java/net/server/channel/packet/special/MonsterMagnetPacket.java
package net.server.channel.packet.special;

import java.awt.Point;

public class MonsterMagnetPacket extends BaseSpecialMovePacket {
   private final MonsterMagnetData[] monsterData;

   private final Byte direction;

   public MonsterMagnetPacket(Integer skillId, Integer skillLevel, Point position, MonsterMagnetData[] monsterData, Byte direction) {
      super(skillId, skillLevel, position);
      this.monsterData = monsterData;
      this.direction = direction;
   }

   public MonsterMagnetData[] monsterData() {
      return monsterData;
   }

   public Byte direction() {
      return direction;
   }
}
